src/SFA.DAS.Commitments.Infrastructure/Data/ApprenticeshipRepository.cs

```diff
@@ -630,30 +631,50 @@ public async Task<IEnumerable<ApprenticeshipStatusSummary>> GetApprenticeshipSum
             });
         }
 
-        public async Task<ApprenticeshipsResult> GetActiveApprenticeshipsByProvider(long providerId)
+        public async Task<ApprenticeshipsResult> GetApprovedApprenticeshipsByEmployer(long accountId)
         {
-            return await GetActiveApprenticeships("[GetActiveApprenticeshipsForProvider]", providerId);
+            return await GetApprovedApprenticeships("GetApprovedApprenticeshipsForEmployer", accountId);
         }
 
-        public async Task<ApprenticeshipsResult> GetActiveApprenticeshipsByEmployer(long accountId)
+        public async Task<ApprenticeshipsResult> GetApprovedApprenticeshipsByProvider(long accountId)
         {
-            return await GetActiveApprenticeships("[GetActiveApprenticeshipsForEmployer]", accountId);
+            return await GetApprovedApprenticeships("GetApprovedApprenticeshipsForProvider", accountId);
         }
 
-        private Task<ApprenticeshipsResult> GetActiveApprenticeships(string sprocName, long id)
+        private Task<ApprenticeshipsResult> GetApprovedApprenticeships(string sprocName, long id)
         {
             return WithConnection(async c =>
             {
                 var parameters = new DynamicParameters();
                 parameters.Add("@id", id, DbType.Int64);
 
-                var apprenticeships = (await c.QueryAsync<Apprenticeship>(sprocName, parameters, commandType: CommandType.StoredProcedure))
-                    .ToList();
-                
+                var apprenticeships = new Dictionary<long, Apprenticeship>();
+
+                using (var multi = await c.QueryMultipleAsync(sprocName, parameters, commandType: CommandType.StoredProcedure))
+                {
+                    multi.Read<Apprenticeship, DataLockStatusSummary, Apprenticeship>(
```

Comment:
I really like this pattern, my only comment is it;s not testable because it's in the Repository and we don't test this layer. I hadn't seen this before

Reply:
True, but none of the code in the repository layer is unit-testable. There is an integration test project that covers it though, along with the acceptance tests.

Reply:
I know, but this extended logic could be, the integration tests will rarely test all the scenarios.